library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -168,23 +163,39 @@ public DownloadBatch retrieveBatchBy(DownloadInfo downloadInfo) {
             batchesCursor.close();
         }
 
-        if (!forDeletion.isEmpty()) {
-            deleteBatchesForIds(forDeletion, downloads);
+        return batches;
+    }
+
+    public void deleteMarkedBatchesFor(Collection<DownloadInfo> downloads) {
+        Cursor batchesCursor = resolver.query(Downloads.Impl.BATCH_CONTENT_URI, null, SELECT_MARKED_FOR_DELETION, null, null);
```

Comment:
http://developer.android.com/reference/android/content/ContentResolver.html#query(android.net.Uri, java.lang.String[], java.lang.String, java.lang.String[], java.lang.String)

Why aren't you using `query` like it should be used? With `selection` and `selectionArgs` like you did originally. What do you gain by combining them? Apart from not following the api :-(